frappe/core/doctype/user/user.py

```diff
@@ -811,6 +812,20 @@ def check_roles_added(self):
 			},
 		)
 
+	def validate_ip_addr(self):
+		# remove whitespace
+		if not self.restrict_ip:
+			return
+		if str.isspace(self.restrict_ip):
+			self.restrict_ip = None
```

Comment:
Comment on taste: Avoid special handling for special cases when it's possible. 

Example from Linus Torvalds: https://github.com/mkirchner/linked-list-good-taste